app/models/driver.rb

```diff
@@ -0,0 +1,30 @@
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :destroy
+  validates :name, presence: true
+  validates :vin, presence: true
+
+  def total_earnings
+    total = 0
+    trips.each do |t|
+        total += t.cost
+    end
+    return (total * 0.85)
+  end
+
+  def ave_rating
```

Comment:
This method always does integer division to calculate the average.  So you can't have a rating of `2.5`.

Also what if they have no trips!  This will generate an error!